google-cloud-firestore/CONTRIBUTING.md

```diff
@@ -1,187 +1,390 @@
+1. [Submit your pull request](#submit-your-pull-request)
+
+## Open an issue
+
+Pull requests should generally be directed by an existing issue, otherwise you risk working on something that the
+maintainers might not be able to accept into the project. Please take a look through [the repository
+issues](https://github.com/googleapis/google-cloud-ruby/issues), and if you do not see an existing issue for your
+problem or feature, please open one using one of the provided templates.
+
+## Sign Contributor License Agreement
+
+Before we can accept your pull requests you'll need to sign a Contributor License Agreement (CLA):
+
+- **If you are an individual writing original source code** and **you own the intellectual property**, then you'll need
+  to sign an [individual CLA](https://developers.google.com/open-source/cla/individual).
+- **If you work for a company that wants to allow you to contribute your work**, then you'll need to sign a [corporate
   CLA](https://developers.google.com/open-source/cla/corporate).
 
-You can sign these electronically (just scroll to the bottom). After that, we'll
-be able to accept your pull requests.
+You can sign these electronically. After that, we'll be able to accept your pull requests.
 
-## Setup
+## Set up environment
 
-In order to use the google-cloud-firestore console and run the project's tests,
-there is a small amount of setup:
+To set up your local development environment:
 
-1. Install Ruby. google-cloud-firestore requires Ruby 2.5+. You may choose to
-   manage your Ruby and gem installations with [RVM](https://rvm.io/),
-   [rbenv](https://github.com/rbenv/rbenv), or
-   [chruby](https://github.com/postmodern/chruby).
+1. Install a [supported version](google-cloud-firestore.gemspec) (or versions) of Ruby. (You may choose to manage your
+   Ruby and gem installations with [RVM](https://rvm.io/), [rbenv](https://github.com/rbenv/rbenv), or
+   [chruby](https://github.com/postmodern/chruby).)
 
-2. Install [Bundler](http://bundler.io/).
+1. Install [Bundler](http://bundler.io/).
 
    ```sh
    $ gem install bundler
    ```
 
-3. Install the top-level project dependencies.
+1. [Fork](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks) the
+   [google-cloud-ruby](https://github.com/googleapis/google-cloud-ruby) repo, clone your fork, and configure the
+   `upstream`
+   [remote](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/configuring-a-remote-for-a-fork_):
+
+   ```bash
+   git clone https://github.com/<your-username>/google-cloud-ruby.git
+   cd google-cloud-ruby
+   git remote add upstream git@github.com:googleapis/google-cloud-ruby.git
+   ```
+
+1. If your fork and clone are not brand new, get the latest changes from `upstream`:
+
+   ```bash
+   git checkout main
+   git pull upstream main
+   ```
+
+1. Install the top-level project dependencies (or update your dependencies in an existing clone):
```

Comment:
Is this step even needed these days? I think the release and CI scripts might use the toplevel bundle, but I don't know that anything else really does, unless you want to run any of the "global" rake tasks.

Reply:
Good catch, will remove.

Reply:
Removed.